### PR TITLE
Sedov Analytical Solution

### PR DESCRIPTION
**Pull request by jamesm2w**
Add in the analytical solution to the Sedov problem.

- `examples/sedov_solution`: Contains a comparison script, as well as generation program, based on the version used in SPH-EXA
- `io/write_hdf5_dump.cpp`: Contains a new output format which looks like something SPH-EXA would generate. 
   - Simulations using `"hdf5_dump"` need a `pressure` scalar force and a `timeStep` variable

### CODE REVIEW

**Comment by Scott943** [comment]
It looks like you've merged the commits from main into this branch (so e.g. all of @mwasacz 's commits from his New API PR are duplicated in this PR). Next time I think if we could just rebase off master, then the commit history will be a little cleaner.

File: src/examples/sedov_solution/main.cpp [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    writeAscii<Real>(0, nSteps, solFile, true, {rSol.data(), rho.data(), u.data(), p.data(), vel.data(), cs.data()},
+                     std::setw(16), std::setprecision(7), std::scientific);
+
+    std::cout << "Created solution file: '" << solFile << std::endl;
+
+    return EXIT_SUCCESS;
+}
+
+void printHelp(char* binName) {
+    printf("\nUsage:\n\n");
```
Comment (Scott943) [inline src/examples/sedov_solution/main.cpp:149]: No idea why the original author used `printf` instead of `std::cout`

**Review by Scott943 — APPROVED** [review]

File: src/examples/sedov_solution/main.cpp (hunk 1 above)
Comment (jamesm2w) [reply src/examples/sedov_solution/main.cpp:149]: Yeah not sure. Tried to not modify most of that code though.

**Review by mwasacz — APPROVED** [review]
I didn't study the code too closely but since it's mostly copy-pasted from SPH-EXA, I think we can merge it.

**Pull request merged**